src/constbuffer_array_splitter.c

```diff
@@ -350,6 +350,28 @@ TARRAY(CONSTBUFFER_ARRAY_HANDLE) constbuffer_array_splitter_split_to_array_of_ar
                             current_buffer_count++;
                             end_buffer_size = max_buffer_size - current_buffer_size;
 
+                            if (current_buffer_size + buffer->size == max_buffer_size)
+                            {
+
+                                while (++i < buffer_count)
+                                {
+                                    CONSTBUFFER_HANDLE next = constbuffer_array_get_buffer(buffers, i);
+                                    const CONSTBUFFER* next_buffer = CONSTBUFFER_GetContent(next);
+                                    if (next_buffer->size == 0)
+                                    {
+                                        current_buffer_count++;
+                                        end_buffer_size = 0;
+                                        CONSTBUFFER_DecRef(next);
+                                    }
+                                    else
+                                    {
+                                        CONSTBUFFER_DecRef(next);
+                                        i--;
+                                        break;
+                                    }
+                                }
+                            }
```

Comment:
Hmmm, I don't see any unit tests being added for this behavior.

Reply:
And I think we should actually have a spec (SRS) that describes what we want.

Reply:
added